pennylane_lightning/core/_measurements_base.py

```diff
@@ -0,0 +1,398 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for state vector measurements.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable, List, Union
+
+import numpy as np
+import pennylane as qml
+from pennylane.devices.qubit.sampling import _group_measurements
+from pennylane.measurements import (
+    ClassicalShadowMP,
+    ExpectationMP,
+    MeasurementProcess,
+    ProbabilityMP,
+    SampleMeasurement,
+    ShadowExpvalMP,
+    Shots,
+    StateMeasurement,
+    VarianceMP,
+)
+from pennylane.ops import Hamiltonian, SparseHamiltonian, Sum
+from pennylane.tape import QuantumScript
+from pennylane.typing import Result, TensorLike
+from pennylane.wires import Wires
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+
+
+class LightningBaseMeasurements(ABC):
+    """Lightning [Device] Measurements class
+
+    A class that serves as a base class for Lightning state-vector simulators.
+    Measures the state provided by the Lightning[Device]StateVector class.
+
+    Args:
+        qubit_state(Lightning[Device]StateVector): Lightning state-vector class containing the state vector to be measured.
+    """
+
+    def __init__(
+        self,
+        qubit_state: Any,
+    ) -> None:
+        self._qubit_state = qubit_state
+
+        # Dummy for the C++ bindings
+        self._measurement_lightning = None
+
+    @property
+    def qubit_state(self):
+        """Returns a handle to the Lightning[Device]StateVector object."""
+        return self._qubit_state
+
+    @property
+    def dtype(self):
+        """Returns the simulation data type."""
+        return self._qubit_state.dtype
+
+    @abstractmethod
+    def _measurement_dtype(self):
+        """Binding to Lightning [Device] Measurements C++ class.
```

Comment:
You use both `Lightning [Device] Measurements` and others `Lightning[Device]Measurements`. let's update this to one version: `Lightning [Device] Measurements` the same for StateVector and Jacobian base classes.